lib/crawler.js

```diff
@@ -1062,98 +1062,97 @@ Crawler.prototype.queueURL = function(url, referrer) {
 
 */
 Crawler.prototype.fetchQueueItem = function(queueItem) {
-    var crawler = this;
-
-    // Variable declarations
-    var clientRequest,
-        timeCommenced;
-
-    // Mark as spooled
-    queueItem.status = "spooled";
-    var client = queueItem.protocol === "https" ? https : http;
-    var agent  = queueItem.protocol === "https" ? crawler.httpsAgent : crawler.httpAgent;
+    var crawler = this,
+        timeCommenced = Date.now();
 
-    // Up the socket limit if required.
-    if (agent.maxSockets < crawler.maxConcurrency) {
-        agent.maxSockets = crawler.maxConcurrency;
-    }
+    crawler.queue.update(queueItem.url, {
+        status: "spooled"
+    }, function(error, queueItem) {
```

Comment:
True, throwing the error is a bit like copping out, I guess. What about the fetching process - do we emit an event and then simply abort the fetching? And leave the queue item as it is?